From fcae4c1383589b5e29b8ae6450be82c06183ee10 Mon Sep 17 00:00:00 2001
From: Yahan Lu <yahan@iscas.ac.cn>
Date: Thu, 5 Jan 2023 07:35:21 +0000
Subject: [PATCH] Revert "[riscv] Remove unnecessary unbound label count"

This reverts commit a6c2b3908073f8ea0c640c205278d017933e78b8.

Reason for revert: Failed tests

Original change's description:
> [riscv] Remove unnecessary unbound label count
>
> The bind_to function doesn't link branch long to trampoline, so it doesn't need to add  unbound_labels_count_.
>
> Change-Id: I2e3861a38eb65c285f19accb12bccb9f4c9fcfb1
> Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4133426
> Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
> Reviewed-by: ji qiu <qiuji@iscas.ac.cn>
> Commit-Queue: ji qiu <qiuji@iscas.ac.cn>
> Cr-Commit-Position: refs/heads/main@{#85103}

Change-Id: I651762d71a8e86bbe76a10224a63433cdacfadfe
No-Presubmit: true
No-Tree-Checks: true
No-Try: true
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4136999
Bot-Commit: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Commit-Queue: Rubber Stamper <rubber-stamper@appspot.gserviceaccount.com>
Cr-Commit-Position: refs/heads/main@{#85108}
---
 src/codegen/riscv/assembler-riscv.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/codegen/riscv/assembler-riscv.cc b/src/codegen/riscv/assembler-riscv.cc
index c7773c57cd..668e3fa9a8 100644
--- a/src/codegen/riscv/assembler-riscv.cc
+++ b/src/codegen/riscv/assembler-riscv.cc
@@ -782,6 +782,10 @@ int32_t Assembler::branch_long_offset(Label* L) {
       L->link_to(pc_offset());
     } else {
       L->link_to(pc_offset());
+      if (!trampoline_emitted_) {
+        unbound_labels_count_++;
+        next_buffer_check_ -= kTrampolineSlotsSize;
+      }
       DEBUG_PRINTF("\tstarted link\n");
       return kEndOfJumpChain;
     }
-- 
2.35.1

